File: webapp-net/TridionDocsMashup/Models/Product.cs
using Sdl.Web.Common.Models;
using System;
using System.Collections.Generic;

namespace Sdl.Web.Modules.TridionDocsMashup.Models
{
    [Serializable]
    [SemanticEntity(EntityName = "Content")]
    public class Product : EntityModel
    {
        [SemanticProperty("_all")]
        public Dictionary<string, KeywordModel> Keywords { get; set; }

        [SemanticProperty("Title")]
        public string Title { get; set; }

        [SemanticProperty("Body")]
        public RichText Body { get; set; }

        [SemanticProperty("Image")]
        public MediaItem Image { get; set; }

    }
}

File: webapp-java/dxa-module-tridion-docs-mashup/src/main/java/com/sdl/dxa/modules/docs/mashup/models/products/Product.java
package com.sdl.dxa.modules.docs.mashup.models.products;

import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.KeywordModel;
import com.sdl.webapp.common.api.model.RichText;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.sdl.webapp.common.api.model.entity.MediaItem;
import java.util.Map;

public class Product extends AbstractEntityModel {

    @SemanticProperty("_all")
    private Map<String, KeywordModel> keywords;

    @SemanticProperty("Title")
    private String title;

    @SemanticProperty("Body")
    private RichText body;

    @SemanticProperty("Image")
    private MediaItem image;

    public Map<String, KeywordModel> getKeywords() {
        return keywords;
    }

    public void setKeywords(Map<String, KeywordModel> keywords) {
        this.keywords = keywords;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) { this.title = title; }

    public RichText getBody() {
        return body;
    }

    public void setBody(RichText body) {
        this.body = body;
    }

    public MediaItem getImage() {
        return image;
    }

    public void setImage(MediaItem image) {
        this.image = image;
    }
}
